src/main/java/com/maxmind/db/BufferHolder.java

```diff
@@ -6,29 +6,27 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.RandomAccessFile;
-import java.nio.ByteBuffer;
 import java.nio.channels.FileChannel;
-import java.nio.channels.FileChannel.MapMode;
 
 final class BufferHolder {
     // DO NOT PASS OUTSIDE THIS CLASS. Doing so will remove thread safety.
-    private final ByteBuffer buffer;
+    private final Buffer buffer;
 
     BufferHolder(File database, FileMode mode) throws IOException {
         try (
             final RandomAccessFile file = new RandomAccessFile(database, "r");
             final FileChannel channel = file.getChannel()
         ) {
             if (mode == FileMode.MEMORY) {
-                final ByteBuffer buf = ByteBuffer.wrap(new byte[(int) channel.size()]);
-                if (channel.read(buf) != buf.capacity()) {
+                final SingleBuffer buf = SingleBuffer.wrap(new byte[(int) channel.size()]);
```

Comment:
It might make sense to move more of this into `SingleBuffer`/`MultiBuffer` as you add `MultiBuffer` given that the handling of the two will need to be very different.

Reply:
Yup, I was planning to do that. :)